tests/integration/preserve_ip_config_test.py

```diff
@@ -0,0 +1,136 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import gi                           # pylint: disable=import-error
+gi.require_version('NM', '1.0')     # NOQA: F402
+from gi.repository import NM        # pylint: disable=import-error
+from libnmstate import netapplier
+
+from .testlib import statelib
+from .testlib import cmd as libcmd
+from .testlib.statelib import INTERFACES
+
+
+_EXTRA_CONFIGS = [
+    {
+        'name': 'ipv4.dad-timeout',
+        'value': '1000'
+    },
+    {
+        'name': 'ipv6.dhcp-hostname',
+        'value': 'libnmstate.example.com'
+    }
+]
+
+# TEST-NET addresses: https://tools.ietf.org/html/rfc5737#section-3
+IPV4_ADDRESS1 = '192.0.2.251'
+# IPv6 Address Prefix Reserved for Documentation:
+# https://tools.ietf.org/html/rfc3849
+IPV6_ADDRESS1 = '2001:db8:1::1'
+
+
+def _get_nm_profile_uuid(iface_name):
```

Comment:
Please use nm.device module for this. You have there `get_device_by_name` and `list_devices`.
Probably you actually just need `connection.get_device_active_connection`.

If you decide to keep this helper, please locate it below the callers.